yew-ui/src/model/decode/peer_decode_manager.rs

```diff
@@ -0,0 +1,120 @@
+use crate::model::MediaPacketWrapper;
+use std::collections::HashMap;
+use std::sync::Arc;
+use types::protos::media_packet::media_packet::MediaType;
+use types::protos::media_packet::MediaPacket;
+use yew::prelude::Callback;
+
+use super::{AudioPeerDecoder, VideoPeerDecoder};
+
+pub struct MultiDecoder {
+    pub audio: AudioPeerDecoder,
+    pub video: VideoPeerDecoder,
+    pub screen: VideoPeerDecoder,
+}
+
+impl MultiDecoder {
+    fn new(video_canvas_id: String, screen_canvas_id: String) -> Self {
+        Self {
+            audio: AudioPeerDecoder::new(),
+            video: VideoPeerDecoder::new(&video_canvas_id),
+            screen: VideoPeerDecoder::new(&screen_canvas_id),
+        }
+    }
+
+    /// Result is:
+    ///     Ok(media_type, Some(true)) on first frame,
+    ///     Ok(media_type, Some(false)) on other frames
+    ///     Ok(media_type, None) on failure to render
+    ///     Err(e) problem with the packet
+    fn decode(&mut self, packet: &Arc<MediaPacket>) -> Result<(MediaType, Option<bool>), i32> {
+        let media_type = packet.media_type.enum_value()?;
+        match media_type {
+            MediaType::VIDEO => Ok((media_type, self.video.decode(packet).ok())),
```

Comment:
is there a way to expose the decode error here? how about: 

```
match media_type {
            MediaType::VIDEO => self.video.decode(packet).map(|_chunk| media_type)
            ...

Reply:
Yeah I'd be happy to -- I was maintaining the previous behavior in which decode errors were ignored.

Reply:
yes, from time to time the encoder craps the bed I forgot about that, whoever calls this method, in this case the component might choose to restart the encoder or to ignore the error.